pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -0,0 +1,562 @@
+	systemNS            = "cattle-system"
+	eksAPIGroup         = "eks.cattle.io"
+	eksV1               = "eks.cattle.io/v1"
+	eksOperatorTemplate = "system-library-rancher-eks-operator"
+	eksOperator         = "rancher-eks-operator"
+	localCluster        = "local"
+	importedAnno        = "eks.cattle.io/imported"
+	awsAccessKey        = "amazonec2credentialConfig-accessKey"
+	awsSecretKey        = "amazonec2credentialConfig-secretKey"
+)
+
+type eksOperatorController struct {
+	clusterEnqueueAfter  func(name string, duration time.Duration)
+	secretsCache         wranglerv1.SecretCache
+	templateLister       mgmtv3.CatalogTemplateLister
+	projectLister        mgmtv3.ProjectLister
+	appLister            projectv3.AppLister
+	appClient            projectv3.AppInterface
+	nsClient             corev1.NamespaceInterface
+	clusterClient        mgmtv3.ClusterInterface
+	systemAccountManager *systemaccount.Manager
+	dynamicClient        dynamic.NamespaceableResourceInterface
+}
+
+func Register(ctx context.Context, wContext *wrangler.Context, mgmtCtx *config.ManagementContext, manager *clustermanager.Manager) {
+	dynamicClient, err := dynamic.NewForConfig(wContext.RESTConfig)
+	if err != nil {
+		panic(fmt.Sprintf("[EKSClusterController Register] error creating dynamic client: %v", err))
+	}
+
+	eksClusterConfigResource := schema.GroupVersionResource{
+		Group:    eksAPIGroup,
+		Version:  "v1",
+		Resource: "eksclusterconfigs",
+	}
+
+	eksCCDynamicClient := dynamicClient.Resource(eksClusterConfigResource)
+
+	e := &eksOperatorController{
+		clusterEnqueueAfter:  wContext.Mgmt.Cluster().EnqueueAfter,
+		secretsCache:         wContext.Core.Secret().Cache(),
+		templateLister:       mgmtCtx.Management.CatalogTemplates("").Controller().Lister(),
+		projectLister:        mgmtCtx.Management.Projects("").Controller().Lister(),
+		appLister:            mgmtCtx.Project.Apps("").Controller().Lister(),
+		appClient:            mgmtCtx.Project.Apps(""),
+		nsClient:             mgmtCtx.Core.Namespaces(""),
+		clusterClient:        mgmtCtx.Management.Clusters(""),
+		systemAccountManager: systemaccount.NewManager(mgmtCtx),
+		dynamicClient:        eksCCDynamicClient,
+	}
+
+	wContext.Mgmt.Cluster().OnChange(ctx, "eks-operator-controller", e.onClusterChange)
+}
+
+func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+	if cluster == nil || cluster.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if cluster.Spec.EKSConfig == nil {
+		return cluster, nil
+	}
+
+	if err := e.deployEKSOperator(); err != nil {
+		return cluster, err
+	}
+
+	if cluster.Status.Driver == "" {
+		cluster = cluster.DeepCopy()
+		cluster.Status.Driver = apimgmtv3.ClusterDriverEKS
+		var err error
+		cluster, err = e.clusterClient.Update(cluster)
+		if err != nil {
+			return cluster, err
+		}
+	}
+
+	eksClusterConfigDynamic, err := e.dynamicClient.Namespace(namespace.GlobalNamespace).Get(context.TODO(), cluster.Name, v1.GetOptions{})
```

Comment:
TODO Context

Reply:
we use this all over to deal with client-go 1.18 requiring contexts when we don't have one to worth passing. Could probably hand one down from Register but that applies to everywhere.